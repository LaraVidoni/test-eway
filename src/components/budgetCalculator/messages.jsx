export default {
  en: {
    budget: "Budget",
    weekly: "Weekly Budget",
    available: "Available",
    timePeriod: "Time Period"
  },
  fr: {
    budget: "Budget",
    weekly: "Budget hebdomadaire",
    available: "disponible",
    timePeriod: "PÃ©riode de temps"
  }
};
