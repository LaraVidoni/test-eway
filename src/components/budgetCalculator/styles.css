.budget-calculator {
  display: flex;
  align-items: center;
  justify-content: center;
  width: auto;
  background-color: #ececec;
  border: 1px solid grey;
  border-radius: 4px;
  height: 28px;
  font-size: 12px;
}
