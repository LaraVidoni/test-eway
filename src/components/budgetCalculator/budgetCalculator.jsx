import React from "react";
import messages from "./messages";
import "./styles.css";

function Price(props) {
  return (
    <span>
      {props.lang === "en" ? "$" + props.amount : props.amount + "$"}{" "}
    </span>
  );
}

function BudgetCalculator(props) {
  let resources = messages[props.language];
  return (
    <div className="budget-calculator">
      <div>
        {props.isWeeklyBudget && resources.weekly || resources.budget} :{" "}
        <Price amount={props.budgetAmount} lang={props.language} />
        {resources.available}: <Price amount={props.availableAmount} lang={props.language} />
        {props.periodEnd &&
          props.periodStart &&
          resources.timePeriod + " : " + props.periodStart + " - " + props.periodEnd}
      </div>
    </div>
  );
}

export default BudgetCalculator;