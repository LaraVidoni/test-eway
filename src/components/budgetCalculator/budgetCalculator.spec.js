import React from 'react';
import {BudgetCalculator} from './budgetCalculator';
import renderer from "react-test-renderer";

it('renders in english (without periods and weekly)', () => {
    const state = {
        isWeeklyBudget : true,
        budgetAmount : "100.00",
        availableAmount : "100.00",
        periodEnd : "",
        periodStart : "",
        language : "en"
    } 
  const div = document.createElement('div');
  const tree = renderer.create(<BudgetCalculator {...state} />, div).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders in english (with periods and weekly)', () => {
    const state = {
        isWeeklyBudget : true,
        budgetAmount : "100.00",
        availableAmount : "100.00",
        periodEnd : "2019-03-22",
        periodStart : "2019-03-22",
        language : "en"
    } 
  const div = document.createElement('div');
  const tree = renderer.create(<BudgetCalculator {...state} />, div).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders in french (without periods and weekly)', () => {
    const state = {
        isWeeklyBudget : true,
        budgetAmount : "100.00",
        availableAmount : "100.00",
        periodEnd : "",
        periodStart : "",
        language : "fr"
    } 
  const div = document.createElement('div');
  const tree = renderer.create(<BudgetCalculator {...state} />, div).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders in french (with periods and weekly)', () => {
    const state = {
        isWeeklyBudget : true,
        budgetAmount : "100.00",
        availableAmount : "100.00",
        periodEnd : "2019-03-22",
        periodStart : "2019-03-22",
        language : "fr"
    } 
  const div = document.createElement('div');
  const tree = renderer.create(<BudgetCalculator {...state} />, div).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders in french (non-weekly)', () => {
    const state = {
        isWeeklyBudget : false,
        budgetAmount : "100.00",
        availableAmount : "100.00",
        periodEnd : "2019-03-22",
        periodStart : "2019-03-22",
        language : "fr"
    } 
  const div = document.createElement('div');
  const tree = renderer.create(<BudgetCalculator {...state} />, div).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders in english (non-weekly)', () => {
    const state = {
        isWeeklyBudget : false,
        budgetAmount : "100.00",
        availableAmount : "100.00",
        periodEnd : "2019-03-22",
        periodStart : "2019-03-22",
        language : "en"
    } 
  const div = document.createElement('div');
  const tree = renderer.create(<BudgetCalculator {...state} />, div).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders in english (huge amount)', () => {
    const state = {
        isWeeklyBudget : false,
        budgetAmount : "999999999999999.00",
        availableAmount : "999999999999999.00",
        periodEnd : "2019-03-22",
        periodStart : "2019-03-22",
        language : "en"
    } 
  const div = document.createElement('div');
  const tree = renderer.create(<BudgetCalculator {...state} />, div).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders in french (huge amount)', () => {
    const state = {
        isWeeklyBudget : false,
        budgetAmount : "999999999999999.00",
        availableAmount : "999999999999999.00",
        periodEnd : "2019-03-22",
        periodStart : "2019-03-22",
        language : "fr"
    } 
  const div = document.createElement('div');
  const tree = renderer.create(<BudgetCalculator {...state} />, div).toJSON();
  expect(tree).toMatchSnapshot();
});