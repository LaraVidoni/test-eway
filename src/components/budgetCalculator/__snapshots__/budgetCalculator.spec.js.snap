// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders in english (huge amount) 1`] = `
<div
  className="budget-calculator"
>
  <div>
    Budget
     :
     
    <span>
      $999999999999999.00
       
    </span>
    Available
    : 
    <span>
      $999999999999999.00
       
    </span>
    Time Period : 2019-03-22 - 2019-03-22
  </div>
</div>
`;

exports[`renders in english (non-weekly) 1`] = `
<div
  className="budget-calculator"
>
  <div>
    Budget
     :
     
    <span>
      $100.00
       
    </span>
    Available
    : 
    <span>
      $100.00
       
    </span>
    Time Period : 2019-03-22 - 2019-03-22
  </div>
</div>
`;

exports[`renders in english (with periods and weekly) 1`] = `
<div
  className="budget-calculator"
>
  <div>
    Weekly Budget
     :
     
    <span>
      $100.00
       
    </span>
    Available
    : 
    <span>
      $100.00
       
    </span>
    Time Period : 2019-03-22 - 2019-03-22
  </div>
</div>
`;

exports[`renders in english (without periods and weekly) 1`] = `
<div
  className="budget-calculator"
>
  <div>
    Weekly Budget
     :
     
    <span>
      $100.00
       
    </span>
    Available
    : 
    <span>
      $100.00
       
    </span>
    
  </div>
</div>
`;

exports[`renders in french (huge amount) 1`] = `
<div
  className="budget-calculator"
>
  <div>
    Budget
     :
     
    <span>
      999999999999999.00$
       
    </span>
    disponible
    : 
    <span>
      999999999999999.00$
       
    </span>
    période de temps : 2019-03-22 - 2019-03-22
  </div>
</div>
`;

exports[`renders in french (non-weekly) 1`] = `
<div
  className="budget-calculator"
>
  <div>
    Budget
     :
     
    <span>
      100.00$
       
    </span>
    disponible
    : 
    <span>
      100.00$
       
    </span>
    période de temps : 2019-03-22 - 2019-03-22
  </div>
</div>
`;

exports[`renders in french (with periods and weekly) 1`] = `
<div
  className="budget-calculator"
>
  <div>
    Budget hebdomadaire
     :
     
    <span>
      100.00$
       
    </span>
    disponible
    : 
    <span>
      100.00$
       
    </span>
    période de temps : 2019-03-22 - 2019-03-22
  </div>
</div>
`;

exports[`renders in french (without periods and weekly) 1`] = `
<div
  className="budget-calculator"
>
  <div>
    Budget hebdomadaire
     :
     
    <span>
      100.00$
       
    </span>
    disponible
    : 
    <span>
      100.00$
       
    </span>
    
  </div>
</div>
`;
